doc_src/fish_mode_prompt.txt

```diff
@@ -2,10 +2,36 @@
 
 \subsection fish_mode_prompt-synopsis Synopsis
 
-fish_mode_prompt will output the mode indicator for use in vi-mode.
+The fish_mode_prompt function will output the mode indicator for use in vi-mode.
 
 \subsection fish_mode_prompt-description Description
 
-The output of `fish_mode_prompt` will be displayed in the mode indicator position to the left of the regular prompt.
+The output of `fish_mode_prompt` will be displayed in the mode indicator
+position to the left of the regular prompt. Define your own function to
+customize the appearance of the mode indicator. The `$fish_bind_mode variable`
+can be used to determine the current mode. It will be one of `default`,
+`insert`, `replace_one`, or `visual`.
```

Comment:
I've addressed all of this in a new commit (`--amend`, `force --push`), except the possibility of mode names changes. I was using an example that used `replace-one` and figured it out simply by `echo`ing the `$fish_bind_mode` variable. I think having potentially out-of-date information is better than nothing at all.